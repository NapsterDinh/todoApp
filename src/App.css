*{
  box-sizing: border-box;
}

body
{
  display: flex;
  justify-content: center;
  align-items: center;
}
.App
{
  
}

.text-center
{
  text-align: center;
}

.App-header
{
  text-align: center;
}

span{
  display: block;
  margin: 0 auto;
  text-align: center;
}

span:not(:first-child)
{
  margin-top: 30px;
}

.input-container
{
  margin-top: 30px;
  display: flex;
  justify-content: center;
  align-items: center;
}
.input-container input
{
  padding: 10px 5px;
  min-width: 300px;
}
.input-container button
{
  background: transparent;
  outline: none;
  border: none;
  color: rgb(23, 84, 252);
  font-size: 32px;
  cursor: pointer;
}

.todo-list
{
  list-style-type: none;
  padding-inline-start: 0;
}

.todo-list li 
{
  position: relative;
  border: 1px solid blue;
  padding: 10px 40px 10px 10px;
  background-color: #DAE8FC;
  margin: 20px auto;
}

.todo-list li.done
{
  background-color: #D5E8D4;
}

.todo-list li span.todo-item-title
{
  text-align: left;
  cursor: pointer;
  max-width: 400px;
  display:-webkit-box;
  -webkit-line-clamp:1;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
  word-break: break-word;
}
.todo-list li span:not(.todo-list li span.todo-item-title)
{
  position: absolute;
  top: -2px;
  right: 10px;
  margin-top: 0;
  font-size: 32px;
  cursor: pointer;
}


