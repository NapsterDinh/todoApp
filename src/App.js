import { useState } from "react";

import Title from "components/Title";
import Input from "components/Input";
import ToDoList from "components/TodoList";
import "App.css";

function App() {
  const [todoList, setToDoList] = useState([]);

  return (
    <div className="App">
      <Title />
      <Input todoList={todoList} setToDoList={setToDoList} />
      <ToDoList todoList={todoList} setToDoList={setToDoList} />
    </div>
  );
}

export default App;
