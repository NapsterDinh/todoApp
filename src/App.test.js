import React from "react";
import App from 'App'
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";

describe('rendering App component correctly', () => {

  it('render App component without crashing', () => {
    shallow(<App/>)
  })
  //Snapshot test will create a component at the firs time execute test
  //and save this component at file JSON. 
  //next time Jest will compare output with this JSON
  it('App matches the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})