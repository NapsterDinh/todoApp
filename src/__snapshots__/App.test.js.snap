// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering App component correctly App matches the snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="App-header text-center"
  >
    To-Do List
  </h1>
  <span>
    Enter text into the input field to add items to your list
  </span>
  <span>
    Click the 'X' to remove the item from your list
  </span>
  <span>
    Click the item to mark it as complete
  </span>
  <div
    className="input-container"
  >
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Input to do"
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
      type="button"
    >
      âŠ•
    </button>
  </div>
  <ul
    className="todo-list"
  />
</div>
`;
